src/panels/lovelace/cards/hui-thermostat-card.ts

```diff
@@ -296,223 +96,56 @@ export class HuiThermostatCard extends LitElement implements LovelaceCard {
-    const stateObj = this.hass!.states[this._config!.entity] as ClimateEntity;
-
-    if (stateObj.attributes.target_temp_step) {
-      return stateObj.attributes.target_temp_step;
-    }
-    return this.hass!.config.unit_system.temperature === UNIT_F ? 1 : 0.5;
-  }
-
-  private _getSetTemp(
-    stateObj: HassEntity
-  ): undefined | number | [number, number] {
-    if (stateObj.state === UNAVAILABLE) {
-      return undefined;
-    }
-
-    if (
-      stateObj.attributes.target_temp_low &&
-      stateObj.attributes.target_temp_high
-    ) {
-      return [
-        stateObj.attributes.target_temp_low,
-        stateObj.attributes.target_temp_high,
-      ];
-    }
-
-    return stateObj.attributes.temperature;
-  }
-
-  private _dragEvent(e): void {
-    const stateObj = this.hass!.states[this._config!.entity] as ClimateEntity;
-
-    if (e.detail.low) {
-      this._setTemp = [e.detail.low, stateObj.attributes.target_temp_high];
-    } else if (e.detail.high) {
-      this._setTemp = [stateObj.attributes.target_temp_low, e.detail.high];
-    } else {
-      this._setTemp = e.detail.value;
+      return html`
+        <hui-warning>
+          ${createEntityNotFoundWarning(this.hass, this._config.entity)}
+        </hui-warning>
+      `;
     }
-  }
 
-  private _setTemperature(e): void {
-    const stateObj = this.hass!.states[this._config!.entity] as ClimateEntity;
+    const name = this._config!.name || computeStateName(stateObj);
 
-    if (e.detail.low) {
-      const newVal = e.detail.low;
-      this._callServiceHelper(
-        stateObj.attributes.target_temp_low,
-        newVal,
-        "set_temperature",
-        {
-          target_temp_low: newVal,
-          target_temp_high: stateObj.attributes.target_temp_high,
-        }
-      );
-    } else if (e.detail.high) {
-      const newVal = e.detail.high;
-      this._callServiceHelper(
-        stateObj.attributes.target_temp_high,
-        newVal,
-        "set_temperature",
-        {
-          target_temp_low: stateObj.attributes.target_temp_low,
-          target_temp_high: newVal,
-        }
-      );
-    } else {
-      const newVal = e.detail.value;
-      this._callServiceHelper(
-        stateObj!.attributes.temperature,
-        newVal,
-        "set_temperature",
-        { temperature: newVal }
-      );
-    }
-  }
+    const color = stateColorCss(stateObj);
 
-  private _renderIcon(mode: string, currentMode: string) {
-    if (!modeIcons[mode]) {
-      return nothing;
-    }
     return html`
-      <ha-icon-button
-        class=${classMap({ "selected-icon": currentMode === mode })}
-        .mode=${mode}
-        @click=${this._handleAction}
-        tabindex="0"
-        .path=${modeIcons[mode]}
-        .label=${this.hass!.localize(
-          `component.climate.entity_component._.state.${mode}`
-        ) || mode}
-      >
-      </ha-icon-button>
+      <ha-card>
+        <p class="title">${name}</p>
+        <ha-more-info-climate-temperature
```

Comment:
```suggestion
        <ha-climate-control
```

Reply:
I was planning to do this in a future PR but ok, let's do this now.

Reply:
future PR is fine too